### PR TITLE
Create 02-spatial-data.Rmd

### PR DESCRIPTION
**Pull request by jannes-m**
Deleting a few typos and again giving a few recommendations (plot(world[0]) and spatial overlay).

### CODE REVIEW

File: 02-spatial-data.Rmd [hunk 1]
```diff
@@ -51,7 +51,7 @@ This can be seen as the final column name of `world`:
 names(world)
 ```
 
-It is the contents of this modest-looking `geom` column, gives `sf` objects their spatial powers.
+It is the contents of this modest-looking `geom` column that gives `sf` objects their spatial powers.
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd]: True that - thanks for the grammar fix!

File: 02-spatial-data.Rmd [hunk 2]
```diff
@@ -145,6 +145,14 @@ For more advanced map making we recommend using a dedicated visualisation packag
 - map export 
 -->
 
+
+<!--Maybe show also somewhere that `world[0]` produces only a plot of the geometry which is rather useful if you do not want to plot a specific attribute. This way, you can for example dismiss the col = "white"-argument in your Nigeria example.
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd]: Agreed - only recently discovered that. Very useful.

File: 02-spatial-data.Rmd [hunk 3]
```diff
@@ -169,10 +177,12 @@ text(x = ncentre_num[1], y = ncentre_num[2], labels = "Nigeria")
 ### Further work
 
 **sf** makes R data objects more closely aligned to the data model used in GDAL and GEOS, in theory making spatial data operations faster.
-The work here provides a taster of the way that **sf** operates but there is much more to learn (see Chapter \@ref(spatial-data-operations)).
-There is a wealth of information that is available in the package's vignettes: these are highly recommended.
+The work here provides a taste of the way how **sf** operates but there is much more to learn (see Chapter \@ref(spatial-data-operations)).
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd]: "the way how" ain't correct grammar. I think it was clumsy though. My suggestion is just to use "the way":

```
The work here provides a taste of the way **sf** operates but there is much more to learn (see Chapter \@ref(spatial-data-operations)).
```

File: 02-spatial-data.Rmd [hunk 4]
```diff
@@ -169,10 +177,12 @@ text(x = ncentre_num[1], y = ncentre_num[2], labels = "Nigeria")
...
+There is a wealth of information in the highly recommended vignettes of the package.
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd]: More concise. :+1:

File: 02-spatial-data.Rmd [hunk 5]
```diff
@@ -169,10 +177,12 @@ text(x = ncentre_num[1], y = ncentre_num[2], labels = "Nigeria")
...
 
 As a final exercise, we'll see how to do a spatial overlay in **sf** by first converting the countries of the world into centroids and then subsetting those in Asia.
+<!-- Maybe rephrase. I was instantly thinking of an actual spatial overlay (`over`) but you mereley refer to a map overlay. -->
```
Comment (Robinlovelace) [inline 02-spatial-data.Rmd]: Agreed.

File: 02-spatial-data.Rmd (hunk 3 above)
Comment (jannes-m) [inline 02-spatial-data.Rmd]: perfect

File: 02-spatial-data.Rmd (hunk 2 above)
Comment (jannes-m) [inline 02-spatial-data.Rmd]: `plot(world$geom)` is even more intuitive than `plot(world[0])`, and exactly doing the same, i.e., just plotting the geometry,

File: 02-spatial-data.Rmd [hunk 6]
```diff
@@ -177,27 +177,30 @@ text(x = ncentre_num[1], y = ncentre_num[2], labels = "Nigeria")
...
-<!-- Maybe rephrase. I was instantly thinking of an actual spatial overlay (`over`) but you mereley refer to a map overlay. -->
+As a final exercise, we'll see one way of how to do a spatial overlay in **sf**. First, we convert the countries of the world into centroids, and then subset those in Asia. Finally, the `summary`-command tells us how many centroids (countries) are part of Asia (43) and how many are not (134).
+
+```{r}
+world_centroids = st_centroid(world)
+sel_asia = st_intersects(world_centroids, asia, sparse = FALSE)
+summary(sel_asia)
+```
+
+Note: another way of achieving the same result is with a GEOS function for identifying spatial overlay, which we'll cover in more datail in Chapter \@ref(spatial-data-operations).
```
Comment (jannes-m) [inline 02-spatial-data.Rmd]: Sorry, my mistake, you are actually performing a spatial overlay but I got confused because you said you were about to do a spatial overlay but then you plot the continents and add the country coordinates. That's why I rearranged the order. I think that makes it clearer.

Comment (Robinlovelace) [inline 02-spatial-data.Rmd]: Yes that makes sense.
